static/js/tools/map/compute_hooks.ts

```diff
@@ -67,3 +77,49 @@ export function useUpdateGeoJson(
     dispatch,
   ]);
 }
+
+export function useComputeBreadcrumbValues(chartStore: ChartStore): {
+  [dcid: string]: number;
+} {
+  const breadcrumbStatReady = useBreadcrumbStatReady(chartStore);
+  const breadcrumbDenomStatReady = useBreadcrumbDenomStatReady(chartStore);
+  const ifRatio = useIfRatio();
+  return useMemo(() => {
+    if (!breadcrumbStatReady()) {
+      return null;
+    }
+    if (ifRatio && !breadcrumbDenomStatReady()) {
+      return null;
+    }
+    console.log("compute breadcrumb values");
+    const breadcrumbValues = {};
+    for (const place in chartStore.breadcrumbStat.data.data) {
+      const placeChartData = getPlaceChartData(
+        chartStore.breadcrumbStat.data.data,
+        place,
+        ifRatio,
+        chartStore.breadcrumbDenomStat.data
+          ? chartStore.breadcrumbDenomStat.data.data
+          : null,
+        Object.assign(
```

Comment:
instead of creating this object for every place in the for loop, can we do it once outside the for loop?